src/main/java/com/example/demo/service/impl/RestoreGeometryDataServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+@Service
+@EnableScheduling
+public class RestoreGeometryDataServiceImpl implements RestoreGeometryDataService {
+    private static final String BACKUP_FILE_NAME = "backup.sql";
+    private static final String RESTORE_FILE_NAME = "restore.sql";
+    private static final String PG_DUMP = "pg_dump";
+    private static final String HOST_CMD = "-h";
+    private static final String PORT_CMD = "-p";
+    private static final String DATABASE_CMD = "-d";
+    private static final String FILE_CMD = "-f";
+    private static final String USER_CMD = "-U";
+    private static final String TABLE_CMD = "-t";
+    private static final String PSQL = "psql";
+    private final AmazonS3Service amazonS3Service;
+    private final String password;
+    private final String user;
+    private final String host;
+    private final String port;
+    private final String dbName;
+
+    public RestoreGeometryDataServiceImpl(AmazonS3Service amazonS3Service,
+                                          @Value("${spring.datasource.password}") String password,
+                                          @Value("${spring.datasource.username}") String user,
+                                          @Value("${spring.datasource.host}") String host,
+                                          @Value("${spring.datasource.port}") String port,
+                                          @Value("${spring.datasource.db.name}") String dbName) {
+        this.amazonS3Service = amazonS3Service;
+        this.password = password;
+        this.user = user;
+        this.host = host;
+        this.port = port;
+        this.dbName = dbName;
+    }
+
+    @Override
+    public boolean restore() {
+        S3Object s3Object = amazonS3Service.getFile(BACKUP_FILE_NAME);
+        File file = new File(RESTORE_FILE_NAME);
+        try (S3ObjectInputStream inputStream = s3Object.getObjectContent()) {
+            Files.copy(inputStream, file.toPath());
+            if (file.exists()) {
+                ProcessBuilder builder = new ProcessBuilder(getRestoreCommand(file.getAbsolutePath()));
+                builder.environment().put("PGPASSWORD", password);
+                return builder.start().waitFor() == 0 && file.delete();
+            }
+            return false;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    @Scheduled(cron = "* */15 * * * *")
+    public void backupGeometries() throws IOException, InterruptedException {
+        File file = new File(BACKUP_FILE_NAME);
+        ProcessBuilder builder = new ProcessBuilder(getBackupCommand(file.getAbsolutePath()));
+        builder.environment().put("PGPASSWORD", password);
+        if (builder.start().waitFor() == 0) {
+            amazonS3Service.save(file);
+            file.delete();
+        }
+        throw new IOException("Can't create backup.sql");
```

Comment:
It is not IOException type.
IOException related to input/output issues. Please read about types of exceptions and select a better one.